modules/graceful-shutdown.adoc

```diff
@@ -35,7 +35,7 @@ $ nodes=$(oc get nodes -o name)
 $ for node in ${nodes[@]}
 do
     echo "==== Shut down $node ===="
-    ssh core@$node sudo shutdown -h 1
+    oc debug $node -- chroot /host sudo shutdown -h 1
```

Comment:
> oc debug $node

Must prefix with 'node/': oc debug node/$node. Otherwise it takes as a Pod name:
$ oc debug ip-10-0-158-180.ap-northeast-2.compute.internal 

As to you hit `error: arguments in resource/name form may not have more than one slash`, I don't know what your command is. I tried to guess and found this reproduces but '//' is of course mistake:
$ oc debug no//ip-10-0-158-180.ap-northeast-2.compute.internal
error: arguments in resource/name form may not have more than one slash
Error from server (NotFound): pods "ip-10-0-158-180.ap-northeast-2.compute.internal" not found

Reply:
In addition, if you use oc debug node/$node -- chroot /host sudo ..., then need to not use sudo because it is already root user. Using sudo for root user does not fail, though.

Reply:
@xingxingxia So when i tested this, the `node/` was already part of the text passed as $node. The list was gathered  using `$ nodes=$(oc get nodes -o name)`.

Did you use that same command to gather the node names? I can test again and make sure I'm seeing the same thing. Perhaps if it's confusing, we could always change that command that gathers the list of node names to not include `node/`, and instead include that as part of this `oc debug` command instead.

Reply:
And good point about the sudo command, I can remove that.

Reply:
OK, prefer to change the node name gathering command to not include node/ in $node

Reply:
@bergerhoffer found new problem if we use `oc debug`:
```
$ oc get no
NAME                                              STATUS   ROLES    AGE   VERSION
ip-10-0-129-<snipped>   Ready    worker   23h   v1.19.0+8a39924
ip-10-0-154-<snipped>   Ready    master   23h   v1.19.0+8a39924
ip-10-0-176-<snipped>   Ready    worker   23h   v1.19.0+8a39924
ip-10-0-178-<snipped>   Ready    master   23h   v1.19.0+8a39924
ip-10-0-217-<snipped>   Ready    master   23h   v1.19.0+8a39924
ip-10-0-221-<snipped>   Ready    worker   23h   v1.19.0+8a39924
$ nodes=$(oc get nodes -o name | sed 's/^node\//')
$ for node in ${nodes[@]}
> do
>     echo "==== Shut down node/$node ===="
>     oc debug node/$node -- chroot /host sudo shutdown -h 1
> done
==== Shut down node/ip-10-0-129-<snipped> ====
...
Shutdown scheduled for Fri 2020-09-25 10:16:06 UTC, use 'shutdown -c' to cancel.
Removing debug pod ...
...
==== Shut down node/ip-10-0-154-<snipped> ====
...
Shutdown scheduled for Fri 2020-09-25 10:16:10 UTC, use 'shutdown -c' to cancel.
Removing debug pod ...
...
==== Shut down node/ip-10-0-176-<snipped> ====
...
Shutdown scheduled for Fri 2020-09-25 10:16:35 UTC, use 'shutdown -c' to cancel.
Removing debug pod ...
...
==== Shut down node/ip-10-0-178-<snipped> ====
...
Shutdown scheduled for Fri 2020-09-25 10:17:00 UTC, use 'shutdown -c' to cancel.
Removing debug pod ...
...
==== Shut down node/ip-10-0-217-<snipped> ====
...
Shutdown scheduled for Fri 2020-09-25 10:17:24 UTC, use 'shutdown -c' to cancel.
Removing debug pod ...
...
==== Shut down node/ip-10-0-221-<snipped> ====
The connection to the server api.<snipped>:6443 was refused - did you specify the right host or port?
$ oc get no
The connection to the server api.<snipped>:6443 was refused - did you specify the right host or port?

Reply:
The cause is, oc debug takes time to create a Running pod on the node but `shutdown -h 1` only waits 1 min. For a cluster with many nodes, when the scheduled shutdowns are already executed on previous nodes, later `oc debug`s are not executed yet.

Reply:
When 2 of 3 masters in the node list are already shutdown, the cluster will be unavailable due to OCP etcd quorum requires 2 of 3 masters must be alive, so `oc debug`s for the later nodes will fail, the later nodes will be still Running. I checked on the cloud console, indeed the last node/ip-10-0-221-<snipped> is still Running

Reply:
Solutions: increase N properly in `shutdown -h N` according to the node number.